iml-services/iml-ntp/src/main.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use futures::TryStreamExt;
+use iml_orm::{
+    alerts::{self, AlertRecordType},
+    hosts::ChromaCoreManagedhost,
+    tokio_diesel::{self, AsyncRunQueryDsl as _, OptionalExtension as _},
+};
+use iml_service_queue::service_queue::consume_data;
+use iml_wire_types::time::State;
+use tracing_subscriber::{fmt::Subscriber, EnvFilter};
+
+pub async fn get_host_by_fqdn(
+    x: &str,
+    pool: &iml_orm::DbPool,
+) -> Result<Option<ChromaCoreManagedhost>, tokio_diesel::AsyncError> {
+    ChromaCoreManagedhost::by_fqdn(x)
+        .first_async(pool)
+        .await
+        .optional()
+}
```

Comment:
It feels like this function should be over in `iml-orm`

Reply:
Maybe... this doesn't really do anything specific from a business-logic perspective; that part is already over in iml-orm.

All it does is asynchronously fetch the first record and allow for it to be empty.